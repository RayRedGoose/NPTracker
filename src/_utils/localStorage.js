export const getItem = item => {
  return JSON.parse(localStorage.getItem(item))
}

export const addItem = (type, item) => {
  return localStorage.setItem(type, JSON.stringify(item))
}

export const addItemToAll = (path, item) => {
  const targetPath = (!getItem(path)) ? [ ] : [...getItem(path)]
  return localStorage.setItem(path, JSON.stringify([...targetPath, item]))
}
