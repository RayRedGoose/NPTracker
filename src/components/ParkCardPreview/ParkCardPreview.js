import './ParkCardPreview.scss'
import React from 'react'
import starEmpty from 'assets/star-empty.svg'
import arrow from 'assets/more.svg'

const ParkCardPreview = ({id, image, name}) => {
  return (
    <section id={id} className="park-card">
      <div className="preview" style={{backgroundImage: `url(${image})`}}>
        <header>
          <img src={starEmpty} alt="empty star"/>
          <img src={arrow} alt="arrow"/>
        </header>
      </div>
      <h2>{ name }</h2>
    </section>
  )
}

export default ParkCardPreview
