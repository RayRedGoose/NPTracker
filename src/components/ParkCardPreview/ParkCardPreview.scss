@import "../../variables.scss";
@import "../../index.scss";

.park-card {
  background-color: $darken-small;
  padding: 20px;
  width: 30%;
  margin-bottom: 20px;
  .preview {
    width: 100%; height: 200px;
    background-position: center;
    background-size: cover;
    background-repeat: no-repeat;
    header {
      background-color: $darken-medium;
      display: flex;
      justify-content: space-between;
      padding: 10px;
      img {
        cursor: pointer;
        width: 20px;
        @extend %click-action;
        margin: 0 10px;
      }
    }
  }
  h2 {
    color: $accent-color-primary-light;
    font-family: "Salsa";
    letter-spacing: 0.1em;
    margin-top: 20px;
    text-align: center;
  }
}
