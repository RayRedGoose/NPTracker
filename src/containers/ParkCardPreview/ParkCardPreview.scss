@import "../../variables.scss";
@import "../../index.scss";

.park-card {
  background-color: $darken-small;
  padding: 20px;
  width: 30%; height: fit-content;
  margin: 0 17px 20px;
  .preview {
    width: 100%; height: 200px;
    background-position: center;
    background-size: cover;
    background-repeat: no-repeat;
    header {
      background-color: $darken-medium;
      display: flex;
      justify-content: space-between;
      padding: 10px;
      section {
        display: flex;
        align-items: center;
      }
      .location {
        display: flex;
        align-items: center;
        @extend %location;
        color: $light-color;
        margin-top: 0;
        margin-right: 20px;
        font-size: 20px;
      }
      img {
        cursor: pointer;
        width: 20px;
        @extend %click-action;
        margin-right: 10px;
      }
    }
  }
  .title {
    color: $accent-color-primary-light;
    display: flex;
    justify-content: center;
    font-family: "Salsa";
    font-size: 24px;
    letter-spacing: 0.1em;
    text-align: center;
    text-decoration: none;
    margin-top: 20px;
    height: 90px;
    &:hover {
      text-shadow: 0 0 10px $accent-color-secondary;
    }
  }
}
