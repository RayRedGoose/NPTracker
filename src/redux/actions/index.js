export const addProcessName = name => ({
  type: 'ADD_PROCESS_NAME',
  name
})

export const addProcessStep = step => ({
  type: 'ADD_PROCESS_STEP',
  step
})

export const increaseStep = () => ({
  type: 'INCREASE_STEP'
})

export const addUserInfo = info => ({
  type: 'ADD_USER_INFO',
  info
})

export const addParks = parks => ({
  type: 'ADD_PARKS',
  parks
})

export const changeActiveTab = id => ({
  type: 'CHANGE_ACTIVE_TAB',
  id
})
