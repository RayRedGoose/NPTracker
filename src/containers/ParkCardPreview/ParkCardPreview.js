import './ParkCardPreview.scss'
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { NavLink } from 'react-router-dom'
import { addPlannedPark, removePlannedPark, removeProcess } from 'redux/actions'
import { addItem, addItemToAll } from '_utils/localStorage'
// import done from 'assets/done.svg'
import bell from 'assets/bell.svg'
import bellFull from 'assets/bell-full.svg'

export class ParkCardPreview extends Component {
  toggleWishLish = () => {
    const { plannedParks, name } = this.props
    return (!plannedParks.includes(name))
      ? this.addToWishList(name)
      : this.removeFromWishList(name, plannedParks)
  }

  addToWishList = (name) => {
    this.props.addPlannedPark(name)
    addItemToAll('planning', name)
  }

  removeFromWishList = (name, plannedParks) => {
    this.props.removePlannedPark(name)
    const newPlans = plannedParks.filter(park => park !== name)
    addItem('planning', newPlans)
  }

  redirect = () => {
    this.props.removeProcess()
  }

  render() {
    const { id, image, fullName, name, states, plannedParks } = this.props
    const title = (fullName.length > 35 ) ? name : fullName
    const link = fullName.toLowerCase().replace(/ /g, '-')
    const pathForLink = `${this.props.type}/${link}`
    const bellType = (plannedParks.includes(name)) ? bellFull : bell
    return (
      <section id={id} className="park-card">
        <div className="preview" style={{backgroundImage: `url(${image})`}}>
          <header>
            <p className="location">{states}</p>
            <section>
              {
                // <img src={done} alt="flag"/>
              }
              <img src={bellType} alt="bell" onClick={this.toggleWishLish}/>
            </section>
          </header>
        </div>
        <NavLink
          to={pathForLink}
          onClick={this.redirect}
          className="title">{ title }</NavLink>
      </section>
    )
  }
}

export const mapStateToProps = ({plannedParks}) => ({
  plannedParks
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    addPlannedPark,
    removePlannedPark,
    removeProcess
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(ParkCardPreview)
