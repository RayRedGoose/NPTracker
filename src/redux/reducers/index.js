import { combineReducers } from 'redux'
import { process } from './singleReducers/process'
import { user } from './singleReducers/user'

const rootReducer = combineReducers({
  process,
  user
})

export default rootReducer
