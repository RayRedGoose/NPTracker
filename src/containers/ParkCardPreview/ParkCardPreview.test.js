import React from 'react'
import { shallow } from 'enzyme'
import { ParkCardPreview, mapStateToProps, mapDispatchToProps } from './ParkCardPreview'
import { addItem, addItemToAll } from '_utils/localStorage'
import { addPlannedPark, removePlannedPark } from 'redux/actions'

jest.mock('_utils/localStorage')

describe("ParkCardPreview", () => {
  let card, instance
  const addPlannedPark = jest.fn()
  const removePlannedPark = jest.fn()
  const mockProps = {
    key: 1,
    id: 1,
    fullName:"Some National Park",
    name: "Some park",
    states:"WA",
    image:"https://www.some.url/preview.jpg",
    plannedParks:["Some park"],
    addPlannedPark,
    removePlannedPark
  }

  beforeEach(() => {
    card = shallow(
      <ParkCardPreview {...mockProps}/>
    )
    instance = card.instance()
  })
  it("should match snapshot with all data passed in correctly", () => {
    expect(card).toMatchSnapshot()
  })

  it("should call toggleWishLish when bell is clicked", () => {
    const spy = jest.spyOn(instance, 'toggleWishLish')
      .mockImplementation(jest.fn())
    instance.forceUpdate()

    card.find('img').simulate('click')

    expect(spy).toHaveBeenCalled()
  })

  it("should call removeFromWishList when toggleWishLish is clicked and park name is in list", () => {
    const spy = jest.spyOn(instance, 'removeFromWishList')
      .mockImplementation(jest.fn())
    instance.forceUpdate()

    instance.toggleWishLish()

    expect(spy).toHaveBeenCalledWith('Some park', ['Some park'])
  })

  it("should call addToWishList when toggleWishLish is clicked and park name is in list", () => {
    const card = shallow(
      <ParkCardPreview {...mockProps} plannedParks={[]}/>
    )
    const instance = card.instance()

    const spy = jest.spyOn(instance, 'addToWishList')
      .mockImplementation(jest.fn())
    instance.forceUpdate()

    instance.toggleWishLish()

    expect(spy).toHaveBeenCalledWith('Some park')
  })

  it("should call addPlannedPark when addToWishList is clicked and park name is in list", () => {

    instance.addToWishList('Some park')

    expect(addPlannedPark).toHaveBeenCalledWith('Some park')
  })

  it("should call addItemToAll when addToWishList is clicked and park name is in list", () => {

    instance.addToWishList('Some park')

    expect(addItemToAll).toHaveBeenCalledWith('planning', 'Some park')
  })

  it("should call removePlannedPark when removeFromWishList is clicked and park name is in list", () => {

    instance.removeFromWishList('Some park', ['Some park'])

    expect(removePlannedPark).toHaveBeenCalledWith('Some park')
  })

  it("should call addItem when addToWishList is clicked and park name is in list", () => {

    instance.addToWishList('Some park')

    expect(addItem).toHaveBeenCalledWith('planning', [])
  })
})

describe("mapStateToProps", () => {
  it("should return the object with correct keys from store", () => {
    const mockStore = {
      plannedParks: ['Some park'],
      activeTab: 2
    }

    const expected = {
      plannedParks: ['Some park']
    }

    const result = mapStateToProps(mockStore)

    expect(result).toEqual(expected)
  })
})

describe("mapDispatchToProps", () => {
  let mockDispatch, mockProps

  beforeEach(() => {
    mockDispatch = jest.fn()
    mockProps = mapDispatchToProps(mockDispatch)
  });

  it("should call dispatch with addPlannedPark after addPlannedPark prop is called", () => {
    const actionToDispatch = addPlannedPark('Some park')

    mockProps.addPlannedPark('Some park')

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it("should call dispatch with removePlannedPark after removePlannedPark prop is called", () => {
    const actionToDispatch = removePlannedPark('Some park')

    mockProps.removePlannedPark('Some park')

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})
