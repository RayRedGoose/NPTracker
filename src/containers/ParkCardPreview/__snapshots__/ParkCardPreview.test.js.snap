// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParkCardPreview should match snapshot with all data passed in correctly 1`] = `
<section
  className="park-card"
  id="1A"
>
  <div
    className="preview"
    style={
      Object {
        "backgroundImage": "url(https://www.some.url/preview.jpg)",
      }
    }
  >
    <header>
      <p
        className="location"
      >
        WA
      </p>
      <section>
        <img
          alt="bell"
          onClick={[Function]}
          src="bell-full.svg"
        />
      </section>
    </header>
  </div>
  <NavLink
    className="title"
    onClick={[Function]}
    to="parks/some-national-park"
  >
    Some National Park
  </NavLink>
</section>
`;
